### PR TITLE
register-component: Remove register component

### PR DESCRIPTION
**Pull request by adamdmharvey**
## Hey, I just made a Pull Request!

Removes the `register-component` plugin.

It seems that only the Register API on the API page button still relied on the `register-component` plugin, and it's route `/register-component`. Everything else was using `/catalog-import`.

image

This is an attempt to deprecate this plugin, and rewire the API button. I suppose it's less of a deprecation, and more of a decommission, so welcome feedback if we may need a more smooth transition here (a la stability index, which already stated it WOULD be removed), or just _poof_, and it's gone is OK.

Note, I was unable to fully test a newly created app, as it seems even off a raw `npx @backstage/create-app` my `/catalog-import` route isn't working, so I think there's something else already broken (which could be fixed separately?) than this PR, though welcome feedback on that. (or if that should be fixed before this)

If this does get merged, I don't know if there's a manual step to unregister from npmjs.com ?

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
    - Avoided a changeset for the `example-app`; is that correct?
    - Did not include a changeset for the app I actually removed, since it of course won't get bumped or have a CHANGELOG any longer
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by adamdmharvey** [comment]
Opened as draft and welcome some review, because a) I'm not quite sure removing this was OK yet, and b) the impact to the create-app. But welcome feedback for tweaking.

**Comment by adamdmharvey** [comment]
Oh and another question - should I manually hack the `yarn.lock` to remove it or just run a new `yarn` and commit the result, does that clean it?

File: plugins/api-docs/src/components/ApiExplorerPage/ApiExplorerPage.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const ApiExplorerPage = () => {
             variant="contained"
             color="primary"
             component={RouterLink}
-            to="/register-component"
+            to="/catalog-import"
```
Comment (adamdmharvey) [inline plugins/api-docs/src/components/ApiExplorerPage/ApiExplorerPage.tsx:40]: There may be a more clean way to dynamically infer the route now (saw additions being added I think to let them be discovered more cleanly?), but for now kept with the hardcoded one like was already there

**Comment by adamdmharvey** [comment]
The `backstage/packages/app/src/App.tsx` is quite different than `backstage/packages/create-app/templates/default-app/packages/app/src/App.tsx` - beyond just plugins, but rather how the page is wired.

For those that know more about it: Does it make sense to use this PR to refactor that, or should I do a separate PR in front of this to refactor it? I assume they should be closer in alignment (just without Lighthouse and some of the example plugins?).

image

**Comment by benjdlambert** [comment]
I think we probably want two PR's.

A release to the `register-component` plugin that would allow us to add the deprecation notice to `yarn installs` pointing users at the `catalog-import` plugin instead. And then a followup PR to remove it from the repo.

**Comment by adamdmharvey** [comment]
@benjdlambert Makes sense!

With respect to the stability index for this plugin which is a 0, that would indicate just a CHANGELOG entry.  Suppose we can just do it in two PRs for ability to do it smoothly, and in case plugins are using it?

https://backstage.io/docs/overview/stability-index#register-component-githubhttpsgithubcombackstagebackstagetreemasterpluginsregister-component

I'll close this and replace with a deprecation one that updates the API link.  I played around with trying to fire a Winston warn in the `register-component`, but couldn't figure out a good way to do it through only the React `src/components/` because there's really no business logic in this plugin to throw warnings in - so welcome any suggestions there! (and couldn't find any examples elsewhere in the project to sort of learn from)

**Comment by regicsolutions** [comment]
@adamdmharvey @benjdlambert do you know if the catalog-import plugin works with Bitbucket? Looks to only work with Github.

I am using the register-component plugin to register components from Bitbucket at the moment and its working fine. Any idea what happened to the register existing component link?

**Comment by adamdmharvey** [comment]
@regicsolutions That's a great question, and I don't officially!  Looking at it, it was written to either a) load an existing YAML from a git location or b) open a pull request.  I'd guess the pull request part likely does not support BitBucket based on it using a GitHub library to open the PR, but since it just leverages other parts of the catalog if you paste an existing YAML in.   If you give it a shot and it works, likely then the only enhancement missing for it to support BitBucket might be to auto detect if the user is trying a PR for GitHub or BitBucket, and open the appropriate one?

**Comment by adamdmharvey** [comment]
and a cc to @mcalus3 who may know more a bit on the background...

**Comment by mcalus3** [comment]
@regicsolutions register-component plugin source code was moved into the catalog-import plugin so it should still support all the previous use cases. If it's not working for registering components from the config hosted on bitbucket, you can open an issue with a bug. I'll look into it.

**Comment by regicsolutions** [comment]
@mcalus3 @adamdmharvey i confirmed the same basic import functionality is working with Bitbucket Server. I do think we need to make the verbiage of the plugin a bit more generic and ideally specific to the integration in use given in the app-config, given someone using Bitbucket wouldn't benefit from all the GitHub verbiage 😀 

For example, import into Backstage should be changed to Import into "my app" and anything that makes reference to GitHub maybe keep it generic for now like from "your repository" until we can have it changed to integration.bitbucket = Bitbucket or integration.gitlab = GitLab etc... thoughts?

**Comment by adamdmharvey** [comment]
@regicsolutions spun out to it's own issue.  the nomenclature ones would be good ones for new comers to pick up, and indeed adding more context fully around multiple Git providers would be powerful.  thanks for the report!

**Pull request closed without merging**